### PR TITLE
Doischer/3108 add contract change notification

### PR DESCRIPTION
**Pull request by tomdoischer**
https://redmine.czechidm.com/issues/3108

### CODE REVIEW

File: Realization/backend/idm-extras/src/main/java/eu/bcvsolutions/idm/extras/scheduler/task/impl/ContractChangeNotificationTask.java [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+			List<IdmAuditDto> audits = auditService.find(auditFilter, null).getContent();
+			for (IdmAuditDto audit : audits) {
+				audit = auditService.get(audit.getId());
+				if (audit.getEntityId() != null) {
+					IdmFormValueDto formValue = formValueService.get(audit.getEntityId());
+					IdmFormAttributeDto formAttribute = DtoUtils.getEmbedded(formValue, IdmFormValue_.FORM_ATTRIBUTE, IdmFormAttributeDto.class);
+					
+					if (contractAttributesEavFormAttrs.contains(formAttribute) && audit.getSubOwnerId() != null && 
+							!changedContractIds.contains(UUID.fromString(audit.getSubOwnerId()))) {
+						changedContractIds.add(UUID.fromString(audit.getSubOwnerId()));
```
Comment (kucera-roman) [inline Realization/backend/idm-extras/src/main/java/eu/bcvsolutions/idm/extras/scheduler/task/impl/ContractChangeNotificationTask.java:193]: Same as comment above, It's a set do we need to check?

File: Realization/frontend/czechidm-extras/src/locales/cs.json [hunk 2]
```diff
@@ -575,6 +575,61 @@
...
+        "label": "Notifikace o změně pracovního úvazku",
+        "help": "Úloha odesílá notifikace o změně pracovního úvazku.",
+        "attributes": {
+          "daystolookback": {
+            "label": "Kolik dní do minulosti jsou změny sledovány"
+          },
+          "contractattributes": {
+            "label": "Standardní atribury úvazku, jejichž změna je sledována"
+          },
+          "contractattributeseav": {
```
Comment (kucera-roman) [inline Realization/frontend/czechidm-extras/src/locales/cs.json:588]: Maybe add info about the format for multiple values?

File: Realization/frontend/czechidm-extras/src/locales/cs.json [hunk 3]
```diff
@@ -575,6 +575,61 @@
...
         }
       },
+      "extras-contract-change": {
+        "label": "Notifikace o změně pracovního úvazku",
+        "help": "Úloha odesílá notifikace o změně pracovního úvazku.",
+        "attributes": {
+          "daystolookback": {
+            "label": "Kolik dní do minulosti jsou změny sledovány"
+          },
+          "contractattributes": {
```
Comment (kucera-roman) [inline Realization/frontend/czechidm-extras/src/locales/cs.json:585]: Maybe add info about the format for multiple values?

**Review by kucera-roman — COMMENTED** [review]
I added some comments.
Otherwise code LGTM.
I like the solution with builder

File: Realization/backend/idm-extras/src/main/java/eu/bcvsolutions/idm/extras/scheduler/task/impl/ContractStartNotificationTask.java [hunk 4]
```diff
@@ -239,7 +239,7 @@ public Optional<OperationResult> processItem(IdmIdentityContractDto dto) {
 				AccAccountFilter accountFilter = new AccAccountFilter();
 				accountFilter.setSystemId(system);
 				accountFilter.setIdentityId(identity.getId());
-				long count = accountService.count(accountFilter, null);
+				long count = accountService.count(accountFilter);
```
[the remark this answers is not in the document]
Comment (tomdoischer) [reply Realization/backend/idm-extras/src/main/java/eu/bcvsolutions/idm/extras/scheduler/task/impl/ContractStartNotificationTask.java:242]: Sorry about this, it changes nothing and is in a different class. But the null argument shouldn't be used when it's optional. IDE yells about that at me so I thought I might fix this when I see it.

[the remark this answers is not in the document]
Comment (kucera-roman) [reply Realization/backend/idm-extras/src/main/java/eu/bcvsolutions/idm/extras/scheduler/task/impl/ContractStartNotificationTask.java:242]: It's ok, I didn't mean it in a bad way. I was just curious why there is change in different task :)

**Review by kucera-roman — APPROVED** [review]
LGTM, thx

**Pull request merged**